File: ios/Classes/platform_api/ad_event_dispatcher_facade/AdEventDispatcherFacade.swift
//
//  AdEventDispatcher.swift
//  device_info_plus
//
//  Created by apple on 30.12.2022.
//

import Foundation

class AdEventDispatcherFacade {
    private let uid: String
    private var dispatcher: AdEventDispatcher
    
    init(uid: String, dispatcher: AdEventDispatcher) {
        self.uid = uid
        self.dispatcher = dispatcher
    }
    
    func sendEvent(type: String, parameters: [String: Any?]?) {
        dispatcher.sendEvent(
            event: AdEvent(uid: uid, type: type, parameters: parameters)
        )
    }
}

File: android/src/main/kotlin/ru/kardenvan/flutter_yandex_ads_plus/platform_api/ad_event_dispatcher_facade/AdEventDispatcherFacade.kt
package ru.kardenvan.flutter_yandex_ads_plus.platform_api.ad_event_dispatcher_facade

import ru.kardenvan.flutter_yandex_ads_plus.platform_api.ad_event.AdEvent
import ru.kardenvan.flutter_yandex_ads_plus.platform_api.ad_event_dispatcher.FlutterYandexAdsEventDispatcher

abstract class AdEventDispatcherFacade(
    /**
     * Unique identifier of the ad
     */
    private val uid: String,

    /**
     * Platform-to-Flutter event dispatcher
     */
    private val dispatcher: FlutterYandexAdsEventDispatcher,
) {
    fun sendEvent(type: String, parameters: Map<String, Any?>? = null) {
        dispatcher.sendEvent(AdEvent(uid, type, parameters))
    }
}